### DISCUSSION TITLE
Various audio filters

### DISCUSSION THREAD

**Discussion by Purfview** [Purfview/whisper-standalone-win#178] [Announcements]
## Various audio filters:

`--ff_dump` 
`--ff_track`
`--ff_fc`
`--ff_mdx_kim2` - It's available only in Faster-Whisper-XXL.
`--ff_mp3`     
`--ff_sync`      
`--ff_rnndn_sh` 
`--ff_rnndn_xiph`
`--ff_fftdn`    
`--ff_tempo`     
`--ff_gate`     
`--ff_speechnorm`
`--ff_loudnorm` 
`--ff_silence_suppress` 
`--ff_lowhighpass`

## Notes:
Needs FFmpeg.
These filters shouldn't be used in Subtitle Edit lower than v4.0.5, because lower versions of SE are using low quality audio input.
`--ff_silence_suppress`'s internal command can't be longer than  32768 chars [explanation is [there](https://stackoverflow.com/questions/2381241/what-is-the-subprocess-popen-max-length-of-the-args-parameter)].

**Comment by ghost** [comment]
--ff_rnndn_sh
is thje goat ig

**Reply by Purfview** [reply]
I find "sh" best too, allegedly "xiph" should be stronger but most of the time "sh" is, on some audios "xiph" looks stronger.
Anyway, you can chain them together, and even add "silence_suppress" too.

For example:
`--ff_rnndn_sh --ff_rnndn_xiph --ff_silence_suppress -40 2`

**Reply by ghost** [reply]
hmm, i use this for everyday use for like movies podcast yt video etc it works pretty well:
```command = [whisper_faster_path, file_path, "--sentence", "--model=large-v2", "-pp", "--compute_type=bfloat16", "--ff_rnndn_sh", "--beep_off", "--batch_recursive"]```
do you recommend to change it with this
`--ff_rnndn_sh --ff_rnndn_xiph --ff_silence_suppress -40 2`

**Reply by Purfview** [reply]
> do you recommend to change it with this
--ff_rnndn_sh --ff_rnndn_xiph --ff_silence_suppress -40 2

I recommend to test it yourself. 😉

On my limited tests I've seen improvement.
Btw, I've seen worse results on audio parts with songs-music when using those rnndn models, but that is expected as singing is not exactly "speech".

**Reply by ghost** [reply]
alrr will test it thanks for improvements bro

**Reply by Purfview** [reply]
I've noticed that "sh" suppress singing when "xiph" kept it, generally "sh" is stronger at suppressing various noises.
Btw, "sh" is abbreviation for "somnolent hogwash". [it doesn't mean anything]

**Reply by ghost** [reply]
@Purfview some podcasts have songs at the beginning, for example transcription didn't have them, I'm quite happy with sh because that's what I want, but first I need a linux build to try xiph :D

**Reply by Purfview** [reply]
These rnndn filters are very fast but not so good for the today's standards, nothing to write home about when compared to the state of art MDX-Net vocal extraction models.
